pkg/subroutines/creator.go

```diff
@@ -0,0 +1,97 @@
+package subroutines
+
+import (
+	"context"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/openmfp/golang-commons/controller/lifecycle"
+	"github.com/openmfp/golang-commons/errors"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1alpha1 "github.com/openmfp/account-operator/api/v1alpha1"
+)
+
+type CreatorSubroutine struct {
+	client  openfgav1.OpenFGAServiceClient
+	storeID string
+}
+
+func NewCreatorSubroutine(fgaGrpcAddr, storeId string) *CreatorSubroutine {
+	conn, _ := grpc.NewClient(fgaGrpcAddr,
+		grpc.EmptyDialOption{},
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithStatsHandler(otelgrpc.NewClientHandler()),
+	)
+
+	cl := openfgav1.NewOpenFGAServiceClient(conn)
```

Comment:
In order to make sure, that your code is properly testable I think you need to adjust the way on how the subroutine is created. Rather than accepting the `fgaGrpcAddr` you would need to accept a version of the `openfgav1.OpenFGAServiceClient` interface, so that in a testing scenario this dependency can be mocked and injected.